x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/support/UserRoleMapper.java

```diff
@@ -222,29 +276,28 @@ public boolean test(FieldExpression.FieldValue fieldValue) {
                     final String suffix = pattern.substring(2);
                     // if the suffix has a wildcard, then it's not a pure sub-tree match
                     if (suffix.indexOf('*') == -1) {
-                        final DN dnSuffix = parseDn(suffix);
-                        if (dnSuffix != null && dn.isDescendantOf(dnSuffix, false)) {
+                        final String normalizedDnSuffix = dnNormalizer.normalize(suffix);
+                        if (normalizedDnSuffix != null && normalizedDn.endsWith("," + normalizedDnSuffix)) {
                             return true;
                         }
                     }
                 }
 
                 return false;
             }
-            if (fieldValue.getValue() instanceof String) {
-                final String testString = (String) fieldValue.getValue();
+            if (fieldValue.getValue()instanceof final String testString) {
```

Comment:
```suggestion
            if (fieldValue.getValue() instanceof final String testString) {
```
? Does spotless not expect a space here?

Reply:
No. It's a tracked bug https://github.com/elastic/elasticsearch/issues/82794

Reply:
😿